api/registrations/views.py

```diff
@@ -282,25 +283,70 @@ def get_object(self):
         return registration
 
 
-class RegistrationContributorsList(NodeContributorsList, RegistrationMixin):
+class RegistrationContributorsList(BaseContributorList, RegistrationMixin, UserMixin):
+    """List of contributors for a registration."""
     view_category = 'registrations'
     view_name = 'registration-contributors'
+    serializer_class = RegistrationContributorsSerializer
 
-    def get_serializer_class(self):
-        return RegistrationContributorsSerializer
+    required_read_scopes = [CoreScopes.NODE_CONTRIBUTORS_READ]
+    required_write_scopes = [CoreScopes.NODE_CONTRIBUTORS_WRITE]
 
+    permission_classes = (
+        ContributorDetailPermissions,
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        ReadOnlyIfRegistration,
+        base_permissions.TokenHasScope,
+    )
 
-class RegistrationContributorDetail(NodeContributorDetail, RegistrationMixin):
+    def get_default_queryset(self):
+        node = self.get_node(check_object_permissions=False)
+        visible_contributors = set(node.visible_contributor_ids)
+        contributors = []
+        index = 0
+        for contributor in node.contributors:
+            contributor.index = index
+            contributor.bibliographic = contributor._id in visible_contributors
+            contributor.permission = node.get_permissions(contributor)[-1]
+            contributor.node_id = node._id
+            contributors.append(contributor)
+            index += 1
+        return contributors
+
+
+class RegistrationContributorDetail(BaseContributorDetail, RegistrationMixin, UserMixin):
+    """Detail of a contributor for a registration."""
     view_category = 'registrations'
     view_name = 'registration-contributor-detail'
     serializer_class = RegistrationContributorsSerializer
 
+    required_read_scopes = [CoreScopes.NODE_CONTRIBUTORS_READ]
+    required_write_scopes = [CoreScopes.NODE_CONTRIBUTORS_WRITE]
 
-class RegistrationChildrenList(NodeChildrenList, RegistrationMixin):
+    permission_classes = (
+        ContributorDetailPermissions,
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        ReadOnlyIfRegistration,
+        base_permissions.TokenHasScope,
+    )
+
+class RegistrationChildrenList(JSONAPIBaseView, generics.ListAPIView, ODMFilterMixin, RegistrationMixin):
+    """Lists the children of a registration."""
```

Comment:
Everything that's appropriate from the original view docstring should still be moved over to this view so that the swagger/BAPI docs are complete. It's not going to be super-DRY, but we are stuck with what we have for docs right now.